src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.ComponentModel.Composition;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.ProjectSystem.VS.ConnectionPoint;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation
+{
+    [Export(typeof(Imports))]
+    [Export(typeof(ImportsEvents))]
+    [AppliesTo(ProjectCapability.VisualBasic)]
+    internal class VSImports : ConnectionPointContainer,
```

Comment:
This class works off the "live" MSBuild object. Is this how other VS project's implementations work? This can basically change under the consumers as they are working with the object, ie writing a loop that reads Count and then reads an index, can cause them to either index out of range at certain points.